environs/config/config.go

```diff
@@ -595,6 +601,23 @@ func Validate(cfg, old *Config) error {
 			return fmt.Errorf("Invalid value for container-networking-method - %v", v)
 		}
 	}
+
+	if v, ok := cfg.defined[CloudInitUserDataKey].(string); ok && v != "" {
```

Comment:
See below - we should do the yaml unmarshall here

Reply:
updated